dashboard/visualization/z_score_plot.py

```diff
@@ -0,0 +1,171 @@
+import numpy as np
+import pandas as pd
+import plotly.graph_objs as go
+
+PLOTLY_TEMPLATE = "plotly_white"
+
+
+def prepare_data_plot(df, well, col, id_pos="id_pos", range_max=None):
+    df = df.groupby(well)[col].agg(["mean", "std"]).reset_index()
+    if df["std"].isna().any():
+        df["std"] = 0.0
+        df["lb"] = df["mean"]
+        df["ub"] = df["mean"]
+    else:
+        df["lb"] = df["mean"] - df["std"]
+        df["ub"] = df["mean"] + df["std"]
+    df = df.sort_values("std")  # can be "mean" as well
+
+    if range_max is not None:
+        x_positions = [0]
+        frac = range_max / len(df)
+        i = frac
+        while i < range_max:
+            x_positions.append(i)
+            i += frac
+        df[id_pos] = x_positions
+    else:
+        df[id_pos] = range(len(df))
+
+    return df
+
+
+def plot_zscore(
```

Comment:
its' quite unintuitive for a plotting function to return a dataframe, moreover the function itself is quite complex
I'd highly recommend breaking it up into at least two parts:
- preparing dataframes for visualization
- making the actual visualization

thus making it less confusing